conans/test/integration/command_v2/search_test.py

```diff
@@ -0,0 +1,216 @@
+import textwrap
+import unittest
+from collections import OrderedDict
+
+from conans.test.utils.tools import TestClient, TestServer
+
+
+class SearchTest(unittest.TestCase):
+
+    def setUp(self):
+        self.servers = OrderedDict()
+        self.servers["remote1"] = TestServer(server_capabilities=[])
+        self.servers["remote2"] = TestServer(server_capabilities=[])
+
+        self.client = TestClient(servers=self.servers)
+
+    def test_search_no_params(self):
+        self.servers = OrderedDict()
+        self.client = TestClient(servers=self.servers)
+
+        self.client.run("search", assert_error=True)
+        self.assertIn("error: the following arguments are required: query", self.client.out)
+
+    def test_search_no_matching_recipes(self):
+        expected_output = ("remote1:\n"
+                           "  There are no matching recipes\n"
+                           "remote2:\n"
+                           "  There are no matching recipes\n")
+
+        self.client.run("search whatever")
+        assert expected_output == self.client.out
+
+    def test_search_no_configured_remotes(self):
+        self.servers = OrderedDict()
+        self.client = TestClient(servers=self.servers)
+
+        self.client.run("search whatever", assert_error=True)
+        self.assertIn("ERROR: The remotes registry is empty", self.client.out)
+
+    def test_search_disabled_remote(self):
+        self.client.run("remote disable remote1")
+        self.client.run("search whatever -r remote1", assert_error=True)
+        self.assertIn("ERROR: Remote 'remote1' is disabled", self.client.out)
+
+
+class SearchRemotesTest(unittest.TestCase):
```

Comment:
For new tests, please use ``pytest`` style.

Reply:
Done